src/types.ts

```diff
@@ -325,11 +333,113 @@ export class TypedValues {
     }
 }
 
+const DECIMAL_REGEX = /^-?\d+(\.\d+)?/;
+const UUID_REGEX = /^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$/;
+
 const parseLong = (input: string|number): Long|number => {
    const long = typeof input === 'string' ? Long.fromString(input) : Long.fromNumber(input);
    return long.high ? long : long.low;
 };
 
+function toLong(value: Long | number): Long {
+    if (typeof value === 'number') {
+        return Long.fromNumber(value);
+    }
+    return value;
+}
+
+function trimFractionalValue(value: string) {
+    if (value.indexOf('.') >= 0) {
+        value = value.replace(/0.*$/, '');
+        if (value.endsWith('.')) {
+            value = value.substr(0, value.length - 1);
+        }
+    }
+    return value;
+}
+
+function toBigInt(high: Long, low: Long): bigint {
```

Comment:
Where does the implementation come from? Could you provide some jsDoc? Same questions about transformations for Decimal and UUID.